src/Components/dashboard/Dashboard.js

```diff
@@ -0,0 +1,156 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { Avatar, Button } from "@mui/material";
+import ViewModuleIcon from "@mui/icons-material/ViewModule";
+import ViewStreamIcon from "@mui/icons-material/ViewStream";
+import PersonIcon from "@mui/icons-material/Person";
+import AddCircleIcon from "@mui/icons-material/AddCircle";
+
+import eventsInformation from "../../utils";
+import EventClickedComponent from "../eventClicked/EventClcked";
+import "./dashboardStyles.scss";
+
+const DashboardComponent = (props) => {
+  const { textAvatar, userName } = props;
+  //Hooks
+  const [viewEvents, setViewEvents] = useState(true);
+  const [eventClicked, setEventClicked] = useState("");
+  const [hiddenDashboard, setHiddenDashboard] = useState(false);
+
+  //functions
+  const changeViewEventsToColumn = () => {
+    setViewEvents(false);
+  };
+
+  const changeViewEventsToRow = () => {
+    setViewEvents(true);
+  };
+
+  const handleButtonEvent = (e) => {
+    let textButton = e.target.innerText;
+
+    if (textButton === "EDIT") {
+      console.log("Editar el evento");
+    } else if (textButton === "JOIN") {
+      e.target.innerText = "LEAVE";
+    } else {
+      e.target.innerText = "JOIN";
+    }
```

Comment:
this logic should not be in here, this logic should be apply when we are looping the events on line 68